yjit/src/backend/ir.rs

```diff
@@ -1507,6 +1517,9 @@ impl Assembler {
 
     pub fn cpush_all(&mut self) {
         self.push_insn(Insn::CPushAll);
+
+        // Disable ccall's assertion. This will be re-enabled by cpop_all.
+        self.set_reg_temps(RegTemps::default());
```

Comment:
I don't understand what this has to do with disabling an assertion?

Reply:
`ccall` [asserts that no stack temps are in registers](https://github.com/ruby/ruby/blob/abafc36ec54e5bbd8f804794b9e582b32a93c79a/yjit/src/backend/ir.rs#L1482). One reason is that those registers get clobbered when the same registers are used for arguments. Another reason is that most C calls end up calling something that triggers GC, so we want to spill everything on stack.

Because `cpush_all` and `cpop_all` recover all registers, you don't need to worry about the first reason. For the second reason, `cpop_all` and `cpush_all` are used only for utility C function calls in utils.rs, so it doesn't trigger GC (I'll leave a comment about it). On `--yjit-dump-insns`, `print_str` is called on every YARV instruction, so I didn't want to use `ctx.spill_temps` to avoid the assertion failure.

Alternatively, we could remove the `print_str` call for `--yjit-dump-insns` and/or the `--yjit-dump-insns` option itself (since `--yjit-dump-disasm` is generally more useful), and just always spill on those `print_*` functions for extra safety against GC.

Reply:
The comment was just hard to make sense of.